hack/lib/version.sh

```diff
@@ -161,8 +161,14 @@ kube::version::ldflags() {
       "-X '${KUBE_GO_PACKAGE}/vendor/k8s.io/component-base/version.${key}=${val}'"
     )
   }
+  DATE=date
+  if which gdate &>/dev/null; then
+      DATE=gdate
+  elif which gnudate &>/dev/null; then
+      DATE=gnudate
+  fi
 
-  add_ldflag "buildDate" "$(date ${SOURCE_DATE_EPOCH:+"--date=@${SOURCE_DATE_EPOCH}"} -u +'%Y-%m-%dT%H:%M:%SZ')"
+  add_ldflag "buildDate" "$(${DATE} ${SOURCE_DATE_EPOCH:+"--date=@${SOURCE_DATE_EPOCH}"} -u +'%Y-%m-%dT%H:%M:%SZ')"
```

Comment:
Unlike `TAR` and `DOCKER` I think `DATE` is more likely to be confused with say, the actual date.
Maybe `GNU_DATE`?